### ISSUE TITLE
DOM order of items in timeline doesn't match visual order

### ISSUE THREAD

**Issue by matt-thompson-rp** [almende/vis#3090]
I'm using the timeline with a custom sort order that sorts items by id. The visual order is correct but the DOM order isn't. I want to style every third item a different color. Because the markup and visual order don't match I can't use CSS nth-child selectors. I have to set the class name every time within the data set.

The issue also exists in the example at http://visjs.org/examples/timeline/items/itemOrdering.html

Item 0 is ordered first but appears as the last item in the markup.

**Comment by yotamberk** [comment]
I'm sorry but I don't understand the problem. Try to explain again (maybe an illustration can help)

**Comment by matt-thompson-rp** [comment]
I created a JSbin to show this easier. 

http://jsbin.com/bawuzi/edit?html,js,console,output

The console shows the data after it has been sorted and right before the timeline is created. The order of items in the console and visually are correct.

However, if I inspect the DOM and look at the contents of ".vis-group", the order doesn't match. See http://imgur.com/Pp1tIRx

"Job Posting" is the second item both in the data set and visually, but it's the first item in div.vis-group

Based on the markup generated, it looks like the code might be outputting the overflow items first followed by those within range regardless of the order.

**Comment by brichbe** [comment]
@matt-thompson-rp Relying on nth-child selectors for timeline item styling seems like a very bad idea.  First of all, the sorted order **_does_** match the visualized order when I view your jsbin in Chrome, so what you're explaining here seems to be browser dependent (that's just one reason why this isn't a VisJs issue...).

Beyond that, there are many situations where you shouldn't assume your DataSet order matches the order in the DOM.  Although you've developed a very small and simple example to illustrate your question, it's not a realistic application.  A realistic timeline application will usually allow the user to move the timeline range backwards and forwards in time, and it would allow the user to change the zoom level.  These types of user actions cause the timeline to be redrawn, which may cause items to be re-stacked and/or removed from the DOM if they're no longer in view.  Any of these types of situations would break your assumption that the DOM order always matches your DataSet order.

Bottom line, [items ](http://visjs.org/docs/timeline/#items) support setting a custom `style` property as well as using `className` styling, so I highly recommend you proceed using those documented methods for styling your app.
